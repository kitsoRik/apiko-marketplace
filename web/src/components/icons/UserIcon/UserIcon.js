import React from 'react';

import './UserIcon.scss';
import { userIconBaseUrl } from '../../../services/api/api';

const UserIcon = ({ src, fullName = "", className, local = false, ...props }) => {

    const initials = fullName.split(" ").map(s => s[0]);

    if (!src) {
        return (
            <div className={`user-icon ${className ?? ""}`} {...props}>
                <span style={{ background: "orange", color: "black" }}>
                    {initials}
                </span>
            </div>
        )
    }

    return (
        <div className={`user-icon ${className ?? ""}`} {...props}>
            <img src={`${local ? "" : userIconBaseUrl}${src}`} alt="User icon" />
        </div>
    );
}

export default UserIcon;