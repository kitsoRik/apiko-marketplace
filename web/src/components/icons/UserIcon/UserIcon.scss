.user-icon {
    height: 40px;
    width: 40px;

    display: flex;
    justify-content: center;
    align-items: center;

    background-color: transparent;
    border: none;

    font-size: 18px;

    & > span {
        width: 100%;
        height: 100%;

        display: flex;
        justify-content: center;
        align-items: center;

        color: #130f02a9;

        border-radius: 50%;

        font-size: 1em;

        cursor: pointer;
        user-select: none;
    }

    & > img {
        width: 100%;
        height: 100%;

        border-radius: 50%;
        overflow: hidden;
    }
}
