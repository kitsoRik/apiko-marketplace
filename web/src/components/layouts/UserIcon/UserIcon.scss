.user-icon {
    height: 48px;
    width: 48px;

    display: flex;
    justify-content: center;
    align-items: center;

    background-color: transparent;
    border: none;

    & > span {
        width: 100%;
        height: 100%;

        display: flex;
        justify-content: center;
        align-items: center;

        color: #130f02A9;

        border-radius: 50%;

        font-size: 18px;

        cursor: pointer;
        user-select: none;
    }

    & > img {
        max-height: 100%;

        border-radius: 50%;
        overflow: hidden;
    }
}