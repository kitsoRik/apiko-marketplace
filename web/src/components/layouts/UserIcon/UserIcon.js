import React from 'react';

import './UserIcon.scss';

const UserIcon = ({ src, fullName = "PR", ...props }) => {

    const initials = fullName.split(" ").map(s => s[0]);

    if(!src) {
        return (
            <div className="user-icon" {...props}>
                <span style={{ background: "orange", color: "black" }}>
                    { initials }
                </span>
            </div>
        )
    }

    return ( 
        <div className="user-icon" {...props}>
            <img src={src} />
        </div>
     );
}

export default UserIcon;