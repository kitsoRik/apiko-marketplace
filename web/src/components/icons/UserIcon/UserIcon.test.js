import React from "react";
import { create } from "react-test-renderer";
import { mount, shallow } from "enzyme";
import UserIcon from ".";

describe("UserIcon", () => {
	it("should render without image", () => {
		const component = create(
			<UserIcon fullName="Pidburachynskyi Rostyslav" />
		);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});
	it("should render with image", () => {
		const component = create(<UserIcon src="http://..." />);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});

	it("should render have text PR", () => {
		const component = mount(
			<UserIcon fullName="Pidburachynskyi Rostyslav" />
		);
		expect(component.find("span").text()).toEqual("PR");
		expect(component.find("img").length).toEqual(0);
	});

	it("should render have img", () => {
		const component = mount(<UserIcon src="http://..." />);
		expect(component.find("span").length).toEqual(0);
		expect(component.find("img")).toBeTruthy();
	});
});
