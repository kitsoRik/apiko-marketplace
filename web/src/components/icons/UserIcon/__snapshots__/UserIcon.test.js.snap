// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserIcon should render with image 1`] = `
<div
  className="user-icon "
>
  <img
    alt="User icon"
    src="http://localhost:3500/api/static/icons/users/http://..."
  />
</div>
`;

exports[`UserIcon should render without image 1`] = `
<div
  className="user-icon "
>
  <span
    style={
      Object {
        "background": "orange",
        "color": "black",
      }
    }
  >
    P
    R
  </span>
</div>
`;
